inputs/gin/parseFunction.go

```diff
@@ -536,33 +585,77 @@ func parseFromCalledFunction(log zerolog.Logger, callExpr *ast.CallExpr, argNo i
 	return nil, false
 }
 
-func parseIdentAndTrace(log zerolog.Logger, argType *ast.Ident, pkgPath string, pkgName string, imports []*ast.ImportSpec, onExtract func(result astUtils.ParseResult)) (error, bool) {
-	assignStmt, ok := argType.Obj.Decl.(*ast.AssignStmt)
-	if !ok {
-		return nil, false
-	}
+// parseIdentAndTrace parses an identifier and traces it back to its definition
+// It returns true if a new pass is needed to parse the function (i.e. it has had to extract another different type)
+// It is designed to match any number of arguments on either side
+func parseIdentAndTrace(s *gengo.Service, log zerolog.Logger, argType *ast.Ident, pkgPath, pkgName, workDir string, imports []*ast.ImportSpec, onExtract func(result astUtils.ParseResult)) (error, bool) {
```

Comment:
Similarly, this function has unnecessary argument as they are contained in gengo.Service

Example of new function argument:

```func parseIdentAndTrace(s *gengo.Service, argType *ast.Ident, pkgPath, pkgName string, imports []*ast.ImportSpec, onExtract func(result astUtils.ParseResult)) (error, bool)```

Reply:
Same as before